test/e2e/suite/trustpolicy/trustpolicy_test.go

```diff
@@ -0,0 +1,467 @@
+
+			notation.ExpectFailure().Exec("verify", artifact.ReferenceWithDigest(), "-v").
+				MatchErrKeyWords("authenticTimestamp validation failed",
+					VerifyFailed)
+		})
+	})
+
+	It("test override permissive signatureVerification level with Authenticity set to log", func() {
+		Host(nil, func(notation *utils.ExecOpts, _ *Artifact, vhost *utils.VirtualHost) {
+			vhost.SetOption(AuthOption("", ""),
+				AddTrustPolicyOption("override_permissive_trustpolicy.json"),
+				AddTrustStoreOption("e2e", filepath.Join(NotationE2ELocalKeysDir, "new_e2e.crt")))
+
+			artifact := GenerateArtifact("e2e-valid-signature", "")
+
+			notation.Exec("verify", artifact.ReferenceWithDigest(), "-v").
+				MatchKeyWords("Warning: authenticity was set to \"log\"",
+					"signature is not produced by a trusted signer",
+					VerifySuccessfully)
+		})
+	})
+
+	It("test invalid trust store", func() {
+		Host(BaseOptions(), func(notation *utils.ExecOpts, _ *Artifact, vhost *utils.VirtualHost) {
+			vhost.SetOption(AddTrustPolicyOption("invalid_trust_store_trustpolicy.json"))
+
+			artifact := GenerateArtifact("e2e-valid-signature", "")
+
+			notation.ExpectFailure().Exec("verify", artifact.ReferenceWithDigest(), "-v").
+				MatchErrKeyWords("authenticity validation failed",
+					"truststore/x509/ca/invalid_store\\\" does not exist",
+					VerifyFailed)
+		})
+	})
+
+	It("test multiple trust stores", func() {
+		Host(nil, func(notation *utils.ExecOpts, artifact1 *Artifact, vhost *utils.VirtualHost) {
+			// artifact1 signed with new_e2e.crt
+			OldNotation(AuthOption("", ""), AddKeyOption("e2e.key", "new_e2e.crt")).
+				Exec("sign", artifact1.ReferenceWithDigest(), "-v").
+				MatchKeyWords(SignSuccessfully)
+
+			// artifact2 signed with e2e.crt
+			artifact2 := GenerateArtifact("e2e-valid-signature", "")
+
+			// setup multiple trust store
+			vhost.SetOption(AuthOption("", ""),
+				AddTrustPolicyOption("multiple_trust_store_trustpolicy.json"),
+				AddTrustStoreOption("e2e-new", filepath.Join(NotationE2ELocalKeysDir, "new_e2e.crt")),
+				AddTrustStoreOption("e2e", filepath.Join(NotationE2ELocalKeysDir, "e2e.crt")))
+
+			notation.WithDescription("verify artifact1 with trust store ca/e2e-new").
+				Exec("verify", artifact1.ReferenceWithDigest(), "-v").
+				MatchKeyWords(VerifySuccessfully)
+
+			notation.WithDescription("verify artifact2 with trust store ca/e2e").
+				Exec("verify", artifact2.ReferenceWithDigest(), "-v").
+				MatchKeyWords(VerifySuccessfully)
+		})
+	})
+
+	It("test valid trusted identity", func() {
+		Host(BaseOptions(), func(notation *utils.ExecOpts, _ *Artifact, vhost *utils.VirtualHost) {
+			vhost.SetOption(AddTrustPolicyOption("valid_trusted_identity_trustpolicy.json"))
+			artifact := GenerateArtifact("e2e-valid-signature", "")
+
+			notation.Exec("verify", artifact.ReferenceWithDigest(), "-v").
+				MatchKeyWords(VerifySuccessfully)
+		})
+	})
+
+	It("test invalid trusted identity", func() {
```

Comment:
Just question, there are multiple ways to make a trusted identity invalid according to our [spec](https://github.com/notaryproject/notaryproject/blob/main/specs/trust-store-trust-policy.md#trusted-identities-constraints) Should we cover all of them or most of them in the e2e test?

Reply:
Added more test cases based on the spec.